spec/forms/flood_risk_engine/add_exemptions_form_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+require_relative "../../support/shared_examples/form_objects"
+
+module FloodRiskEngine
+  RSpec.describe Steps::AddExemptionsForm, type: :form do
+    let(:params_key) { :add_exemptions }
+    let(:enrollment) { FactoryGirl.create(:enrollment) }
+    let(:model_class) { Enrollment }
+    let(:exemptions) { FactoryGirl.create_list(:exemption, 3) }
+    let(:params) { { params_key => { exemption_ids: exemptions.collect(&:id) } } }
+
+    subject { described_class.factory(enrollment) }
+
+    it_behaves_like "a form object"
+
+    it { is_expected.to be_a(described_class) }
+    it { is_expected.to respond_to(:all_exemptions) }
+
+    describe ".save" do
+      it "adds exemptions to enrollment" do
+        expect(enrollment).to receive(:save).and_return(true) # stub save
+
+        subject.validate(params)
+        subject.save
+
+        expect(enrollment.exemptions).to eq(exemptions)
+      end
+    end
+
+    describe ".validate" do
+      context "with empty exemptions params" do
+        let(:params) { { params_key => { exemption_ids: [] } } }
+        it "should fail" do
+          expect(subject.validate(params)).to be(false)
+        end
```

Comment:
Could maybe assert that the correct validation message is there (as specced in prototype powerpoint) ie loaded from a translation file